### PR TITLE
add evaluation flows and question simulation flow

### PR DESCRIPTION
**Pull request by dorisjoy**
# Description
Add below three examples:
1. eval-multi-turn-metrics: provide these metrics (creativity,conversation_quality,answer_relevance,grounding) to measures the quality of a conversation;
flow link: https://ml.azure.com/prompts/flow/0ff0b18c-8d07-4164-b994-f7c6d6e7e8da/f6364f6b-ec50-463e-b956-435725daed62/details?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourcegroups/promptflow/providers/Microsoft.MachineLearningServices/workspaces/promptflow-eastus2&tid=72f988bf-86f1-41af-91ab-2d7cd011db47
metrics result:
image

2. eval-single-turn-metrics: provide these metrics (grounding,answer_relevance,answer_quality,context_precision,answer_similarity,creativity,context_recall,answer_correctness) to measures the quality of a question and answer pair;
flow link: https://ml.azure.com/prompts/flow/0ff0b18c-8d07-4164-b994-f7c6d6e7e8da/fbc8e80f-f367-45bb-9367-66dbeafa6b96/details?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourcegroups/promptflow/providers/Microsoft.MachineLearningServices/workspaces/promptflow-eastus2&tid=72f988bf-86f1-41af-91ab-2d7cd011db47
metrics result:
image

3. question-simulation: used to generate suggestions for the next question based on the previous chat history. When the chat history seems like should be ended, then the flow output will be [STOP].
flow link: https://ml.azure.com/prompts/flow/0ff0b18c-8d07-4164-b994-f7c6d6e7e8da/a770413f-50af-4c27-a299-6d4c043565dd/details?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourcegroups/promptflow/providers/Microsoft.MachineLearningServices/workspaces/promptflow-eastus2&tid=72f988bf-86f1-41af-91ab-2d7cd011db47
question suggestions:
image

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

### CODE REVIEW

**Review by DaweiCai — DISMISSED** [review]

File: examples/flows/standard/simulation-flow/If_continue.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from promptflow import tool
```
Comment (chjinche) [inline examples/flows/standard/simulation-flow/If_continue.py:1]: make the file name startswith small letter?

File: examples/flows/evaluation/eval-flow-for-multi-turn/covert_chat_history_to_conversation.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from promptflow import tool
+
+
+@tool
+def covert_chat_history_to_conversation(chat_history: list) -> dict:
```
Comment (chjinche) [inline examples/flows/evaluation/eval-flow-for-multi-turn/covert_chat_history_to_conversation.py:5]: typo `covert`, should be `convert`.

File: examples/flows/evaluation/eval-flow-for-multi-turn/answer_relevancy.jinja2 [hunk 3]
```diff
@@ -0,0 +1,28 @@
+system:
```
Comment (chjinche) [inline examples/flows/evaluation/eval-flow-for-multi-turn/answer_relevancy.jinja2:1]: relevance?

File: examples/flows/evaluation/eval-flow-for-multi-turn/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Evaluation flow for single turn:
```
Comment (crazygao) [inline examples/flows/evaluation/eval-flow-for-multi-turn/README.md:1]: Please add correct steps for validating these flows, and generate pipelines and general README entry..
I'll check why this patch bypass the check_enforcer.

File: examples/flows/evaluation/eval-flow-for-multi-turn/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+* grounding: Measures whether the answer follows logically from the information contained in the context based on provided answer and context.
+
+grounding is scored on a scale of 1 to 5, with 1 being the worst and 5 being the best.
+
+* answer_relevancy: Measure whether the answer is relevancy to the question based on provided question, context and answer.
+
+answer_relevancy is scored on a scale of 1 to 5, with 1 being the worst and 5 being the best. 
+
+* answer_quality: Measures the answer quality for each of the following factors based on provided question and answer:
```
Comment (chjinche) [inline examples/flows/evaluation/eval-flow-for-multi-turn/README.md:17]: Should the metric be renamed as `conversation_quality`?

File: examples/flows/evaluation/eval-flow-for-multi-turn/answer_relevancy.jinja2 (hunk 3 above)
Comment (dorisjoy) [reply examples/flows/evaluation/eval-flow-for-multi-turn/answer_relevancy.jinja2:1]: Change to answer_relevance

File: examples/flows/evaluation/eval-flow-for-multi-turn/README.md (hunk 5 above)
Comment (dorisjoy) [reply examples/flows/evaluation/eval-flow-for-multi-turn/README.md:17]: Change to conversation_quality

File: examples/flows/evaluation/eval-flow-for-multi-turn/README.md (hunk 4 above)
Comment (dorisjoy) [reply examples/flows/evaluation/eval-flow-for-multi-turn/README.md]: Added workflows for these examples.

File: examples/flows/evaluation/eval-multi-turn-metrics/README.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    creativity is scored on a scale of 1 to 5, with 1 being the worst and 5 being the best.
+
+
+## Prerequisites
+
+- Connection: suggest to use Azure OpenAI or OpenAI connection with 1106 model. If you use the 1106 chat model, then please assign {"type":"json_object"} value to response_format for these nodes: conversation_quality, creativity, answer_relevance.
```
Comment (chjinche) [inline examples/flows/evaluation/eval-multi-turn-metrics/README.md:37]: gpt4?

Comment (chjinche) [reply examples/flows/evaluation/eval-multi-turn-metrics/README.md:37]: > !Note: Recommend to use `gpt-4` series models than the `gpt-3.5` for better performance.

    > !Note: Recommend to use `gpt-4` model (Azure OpenAI `gpt-4` model with version `0613` or later) than `gpt-4-turbo` model (Azure OpenAI `gpt-4` model with version `1106` or later) for better performance. Due to inferior performance of `gpt-4-turbo` model, when you use it, sometimes you might need to set the `response_format`to {"type":"json_object"}, in order to make sure the llm can generate valid json response.

File: examples/flows/evaluation/eval-multi-turn-metrics/conversation_quality_prompt.jinja2 [hunk 7]
```diff
@@ -0,0 +1,40 @@
+system:
+- You are an AI assistent. You will be given a transcript of dialogue between a user and a bot. You need to read the transcript carefully and identify the main topic, question, or issue of the conversation, as well as the purpose and expectations of the interaction.
```
Comment (chjinche) [inline examples/flows/evaluation/eval-multi-turn-metrics/conversation_quality_prompt.jinja2:2]: typo

File: examples/flows/evaluation/eval-single-turn-metrics/calculate_context_recall.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+def calculate(llm_result: str) -> str:
+    try:
+        score = 1
+        response = json.loads(llm_result)
+        if response:
+            result = response.get("result", "")
+            print(result)
+            if result:
+                response = [
+                    int(item.get("attributed", "").lower() == "yes" or item.get("attribited", "").lower() == "yes")
```
Comment (chjinche) [inline examples/flows/evaluation/eval-single-turn-metrics/calculate_context_recall.py:16]: attribited looks typo? Why need to handle?

File: examples/flows/evaluation/eval-multi-turn-metrics/README.md (hunk 6 above)
Comment (dorisjoy) [reply examples/flows/evaluation/eval-multi-turn-metrics/README.md]: Changed.

File: examples/flows/evaluation/eval-single-turn-metrics/calculate_context_recall.py (hunk 8 above)
Comment (chjinche) [reply examples/flows/evaluation/eval-single-turn-metrics/calculate_context_recall.py:16]: Better to add a comment here? Which may help other devs modify this in the future.

Comment (dorisjoy) [reply examples/flows/evaluation/eval-single-turn-metrics/calculate_context_recall.py:17]: Added.

**Review by chjinche — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request closed without merging**